gapis/trace/android/trace.go

```diff
@@ -118,6 +118,26 @@ func (t *androidTracer) GetDevice() bind.Device {
 	return t.b
 }
 
+func (t *androidTracer) ProcessProfilingData(ctx context.Context, buffer *bytes.Buffer) (*service.ProfilingData, error) {
+	// Load Perfetto trace and create trace processor.
+	rawData := make([]byte, buffer.Len())
+	_, err := buffer.Read(rawData)
+	if err != nil {
+		return nil, log.Err(ctx, err, "Failed to read trace buffer")
+	}
+	processor, err := perfetto.NewProcessor(ctx, rawData)
+	defer processor.Close()
+	if err != nil {
+		return nil, log.Errf(ctx, err, "Failed to create trace processor")
+	}
+	gpu := t.b.Instance().GetConfiguration().GetHardware().GetGPU()
+	gpuName := gpu.GetName()
+	if strings.Contains(gpuName, "Adreno") {
+		return adreno.ProcessProfilingData(ctx, processor)
+	}
```

Comment:
I'm not certain this really needs to be GPU specific, but I'm OK to keep it in for now.